src/test/ui/nll/issue-51244.rs

```diff
@@ -3,5 +3,5 @@
 fn main() {
     let ref my_ref @ _ = 0;
     *my_ref = 0;
-    //~^ ERROR cannot assign to `*my_ref` which is behind a `&` reference [E0594]
+    //~^ ERROR cannot assign to `my_ref` which is behind a `&` reference [E0594]
 }
```

Comment:
I "unwrap"ed the `ref` guard(and all the 3 stderr diffs are caused by this) since I felt `*` in  `cannot assign to `*my_ref` which is behind a `&` reference` was a bit not accurate, why not just say `my_ref`? But if it's superfluous, I'll revert the `unwrap`.